From 80fb9609b5077383b6955c76a49281f4fb1369f5 Mon Sep 17 00:00:00 2001
From: Joseph Guo <qijian.guo@nxp.com>
Date: Wed, 25 Jun 2025 10:53:07 +0800
Subject: [PATCH 40/56] LF-15823: arm64: dts: imx95-15x15-frdm: Enable BOE WXGA
 LVDS panel

Enable BOE WXGA LVDS panel for FRDM-IMX95 board.

Signed-off-by: Lei Xu <lei.xu@nxp.com>
Signed-off-by: Joseph Guo <qijian.guo@nxp.com>
Acked-by: Liu Ying <victor.liu@nxp.com>
Reviewed-by: Bough Chen <haibo.chen@nxp.com>
---
 arch/arm64/boot/dts/freescale/Makefile        |  2 +-
 .../imx95-15x15-frdm-boe-wxga-lvds-panel.dts  | 99 +++++++++++++++++++
 2 files changed, 100 insertions(+), 1 deletion(-)
 create mode 100644 arch/arm64/boot/dts/freescale/imx95-15x15-frdm-boe-wxga-lvds-panel.dts

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index 05885b69a44f..123c1640fcf1 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -920,7 +920,7 @@ dtb-$(CONFIG_ARCH_MXC) += imx95-19x19-verdin-rm692c9.dtb
 imx95-19x19-verdin-root-dtbs := imx95-19x19-verdin.dtb imx95-19x19-verdin-root.dtbo
 dtb-$(CONFIG_ARCH_MXC) += imx95-19x19-verdin-root.dtb imx95-19x19-verdin-inmate.dtb
 
-dtb-$(CONFIG_ARCH_MXC) += imx95-15x15-frdm.dtb
+dtb-$(CONFIG_ARCH_MXC) += imx95-15x15-frdm.dtb imx95-15x15-frdm-boe-wxga-lvds-panel.dtb
 
 dtb-$(CONFIG_ARCH_S32) += s32g274a-evb.dtb
 dtb-$(CONFIG_ARCH_S32) += s32g274a-rdb2.dtb
diff --git a/arch/arm64/boot/dts/freescale/imx95-15x15-frdm-boe-wxga-lvds-panel.dts b/arch/arm64/boot/dts/freescale/imx95-15x15-frdm-boe-wxga-lvds-panel.dts
new file mode 100644
index 000000000000..211dc5f1c9a6
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/imx95-15x15-frdm-boe-wxga-lvds-panel.dts
@@ -0,0 +1,99 @@
+// SPDX-License-Identifier: GPL-2.0+
+/*
+ * Copyright 2025 NXP
+ */
+
+#include "imx95-15x15-frdm.dts"
+
+/ {
+	backlight: backlight {
+		compatible = "led-backlight";
+		leds = <&led_backlight1>;
+		brightness-levels = <0 4 8 16 32 64 128 255>;
+		default-brightness-level = <6>;
+	};
+
+	panel-lvds {
+		compatible = "boe,ev121wxm-n10-1850";
+		backlight = <&backlight>;
+		power-supply = <&reg_vcc_12v>;
+		enable-gpios = <&pcal6524 21 GPIO_ACTIVE_HIGH>;
+
+		port {
+			panel_lvds_in: endpoint {
+				remote-endpoint = <&lvds0_out>;
+			};
+		};
+	};
+
+	reg_vcc_12v: regulator-vcc-12v {
+		compatible = "regulator-fixed";
+		regulator-name = "VCC_12V";
+		regulator-min-microvolt = <12000000>;
+		regulator-max-microvolt = <12000000>;
+		gpio = <&pcal6524 22 GPIO_ACTIVE_HIGH>;
+		enable-active-high;
+	};
+};
+
+&display_pixel_link {
+	status = "okay";
+};
+
+&ldb {
+	assigned-clock-rates = <2982000000>, <497000000>;
+
+	channel@0 {
+		status = "okay";
+
+		port@1 {
+			lvds0_out: endpoint {
+				remote-endpoint = <&panel_lvds_in>;
+			};
+		};
+	};
+
+	channel@1 {
+		status = "disabled";
+	};
+};
+
+&ldb0_phy {
+	status = "okay";
+};
+
+&ldb1_phy {
+	status = "disabled";
+};
+
+&lpi2c3 {
+	touchscreen@2a {
+		compatible = "eeti,exc80h60";
+		reg = <0x2a>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_lvds_ctp_int_0>;
+		interrupt-parent = <&gpio5>;
+		interrupts = <1 IRQ_TYPE_LEVEL_LOW>;
+		reset-gpios = <&pcal6524 20 GPIO_ACTIVE_HIGH>;
+		touchscreen-size-y = <4096>;
+		touchscreen-size-x = <4096>;
+	};
+};
+
+&pixel_interleaver {
+	channel@0 {
+		status = "okay";
+	};
+
+	channel@1 {
+		status = "disabled";
+	};
+};
+
+&scmi_iomuxc {
+	pinctrl_lvds_ctp_int_0: lvds_ctp_int_grp_0 {
+		fsl,pins = <
+			IMX95_PAD_XSPI1_DATA1__GPIO5_IO_BIT1    0x31e
+		>;
+	};
+};
-- 
2.34.1

